packages/account-tree-controller/src/AccountTreeController.test.ts

```diff
@@ -1518,19 +1513,28 @@ describe('AccountTreeController', () => {
     });
 
     it('falls back to first wallet first group when AccountsController returns EMPTY_ACCOUNT', () => {
-      const { controller, messenger } = setup({
+      const {
+        controller,
+        messenger,
+        accountsControllerMessenger,
+        accountTreeControllerMessenger,
+      } = setup({
         accounts: [MOCK_HD_ACCOUNT_1, MOCK_HD_ACCOUNT_2],
         keyrings: [MOCK_HD_KEYRING_1, MOCK_HD_KEYRING_2],
       });
 
       // Unregister existing handler and register new one BEFORE init
-      messenger.unregisterActionHandler(
+      accountsControllerMessenger.unregisterActionHandler(
         'AccountsController:getSelectedAccount',
       );
-      messenger.registerActionHandler(
+      accountsControllerMessenger.registerActionHandler(
         'AccountsController:getSelectedAccount',
         () => EMPTY_ACCOUNT_MOCK,
       );
+      messenger.delegate({
+        messenger: accountTreeControllerMessenger,
+        actions: ['AccountsController:getSelectedAccount'],
+      });
```

Comment:
If an action handler is unregistered and then re-registered, the root messenger must delegate the external action to the AccountTreeController messenger again, or the following error will occur: `A handler for AccountsController:getSelectedAccount has not been registered`